src/test/java/org/elasticsearch/index/query/TemplateQueryParserTest.java

```diff
@@ -122,6 +122,74 @@ public void testParser() throws IOException {
         assertTrue("Parsing template query failed.", query instanceof ConstantScoreQuery);
     }
 
+    /**
+     * Test that the template query parser can parse and evaluate template expressed as a single string
+     * with conditional clause.
+     */
+    @Test
+    public void testParseTemplateAsSingleStringWithConditionalClause() throws IOException {
+        String templateString = "{" +
+                "  \"template\" : \"{ \\\"match_{{#use_it}}{{template}}{{/use_it}}\\\":{} }\"," +
+                "  \"params\":{" +
+                "    \"template\":\"all\"," +
+                "    \"use_it\": true" +
+                "  }" +
+                "}";
+
```

Comment:
When trying to figure out what exactly goes wrong when the above query is executed* I noticed that the template string above doesn't look quite right. When changing it to


    String templateString = "{" +
        "  \"template\" : {" +
            " \"query\":{\"match_{{#use_it}}{{template}}{{/use_it}}\":{} }," +
            " \"params\":{" +
                    "    \"template\":\"all\"," +
                    "    \"use_it\": true" +
            "  }" +
            "}";

your test is green on my master - am I missing something here or was that just a misunderstanding on how to formulate the template? Caveat: I didn't check re-formulating the other two tests yet.


* And finally printing the non-quoted version after staring at the quoted one.